### ISSUE TITLE
Rule Change: Change default for `no-useless-computed-key`'s `enforceForClassMembers` option to `true`

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#18042]
### What rule do you want to change?

[no-useless-computed-key](https://eslint.org/docs/latest/rules/no-useless-computed-key)

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/*eslint no-useless-computed-key: "error"*/

class Foo {
    ["foo"] = "bar";

    [0]() {}
    ['a']() {}
    get ['b']() {}
    set ['c'](value) {}

    static ["foo"] = "bar";

    static ['a']() {}
}
```

### What does the rule currently do for this code?

No problem is reported, because the `enforceForClassMembers` defaults to `false`.

### What will the rule do after it's changed?

Report several problems like in [this rule example](https://eslint.org/play/#eyJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSJ9fSwidGV4dCI6Ii8qZXNsaW50IG5vLXVzZWxlc3MtY29tcHV0ZWQta2V5OiBbXCJlcnJvclwiLCB7IFwiZW5mb3JjZUZvckNsYXNzTWVtYmVyc1wiOiB0cnVlIH1dKi9cblxuY2xhc3MgRm9vIHtcbiAgICBbXCJmb29cIl0gPSBcImJhclwiO1xuXG4gICAgWzBdKCkge31cbiAgICBbJ2EnXSgpIHt9XG4gICAgZ2V0IFsnYiddKCkge31cbiAgICBzZXQgWydjJ10odmFsdWUpIHt9XG5cbiAgICBzdGF0aWMgW1wiZm9vXCJdID0gXCJiYXJcIjtcblxuICAgIHN0YXRpYyBbJ2EnXSgpIHt9XG59In0=), but without the option explicitly set.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

This change has been already suggested when the `enforceForClassMembers` option was introduced: https://github.com/eslint/eslint/issues/12041#issuecomment-516802519. Like in other cases, the very reason for having this option was to not introduce a breaking change at that time. This change could be done before v9 is released.

**Comment by mdjermanovic** [comment]
Makes sense to me 👍 

@nzakas what do you think?

**Comment by snitin315** [comment]
Makes sense 👍

**Comment by nzakas** [comment]
Works for me, but don't forget to move the issue through the triage board when responding. :)

**Comment by fasttime** [comment]
Working on this.
